stratum/hal/lib/phal/phal_cli.cc

```diff
@@ -0,0 +1,401 @@
+                std::cout << "Invalid uint32" << std::endl;
+                continue;
+            }
+            update->mutable_value()->set_uint32_val(val);
+            break;
+
+        } else if (!strcmp(type.c_str(), "uint64")) {
+            ::google::protobuf::uint64 val;
+            if (sscanf(val_str.c_str(), "%ld", &val) != 1) {
+                std::cout << "Invalid uint64" << std::endl;
+                continue;
+            }
+            update->mutable_value()->set_uint64_val(val);
+            break;
+
+        } else if (!strcmp(type.c_str(), "double")) {
+            double val;
+            if (sscanf(val_str.c_str(), "%lf", &val) != 1) {
+                std::cout << "Invalid double" << std::endl;
+                continue;
+            }
+            update->mutable_value()->set_double_val(val);
+            break;
+
+        } else if (!strcmp(type.c_str(), "float")) {
+            float val;
+            if (sscanf(val_str.c_str(), "%f", &val) != 1) {
+                std::cout << "Invalid float" << std::endl;
+                continue;
+            }
+            update->mutable_value()->set_float_val(val);
+            break;
+
+        } else if (!strcmp(type.c_str(), "bool")) {
+            if (!strcmp(val_str.c_str(), "false")) {
+                update->mutable_value()->set_bool_val(false);
+                break;
+
+            } else if (!strcmp(val_str.c_str(), "true")) {
+                update->mutable_value()->set_bool_val(true);
+                break;
+
+            } else {
+                std::cout << "Invalid bool <false|true>" << std::endl;
+                continue;
+            }
+
+        } else if (!strcmp(type.c_str(), "string")) {
+            update->mutable_value()->set_string_val(val_str);
+            break;
+
+        } else if (!strcmp(type.c_str(), "bytes")) {
+            update->mutable_value()->set_bytes_val(val_str);
+
+        } else {
+            std::cout << "Must specify a type: <int32, uint32, int64, "
+                      << "uint64, double, float, bool, string, bytes"
+                      << std::endl;
+            continue;
+        }
+    }
+
+    absl::Time start_time = absl::Now();
+    auto status = phaldb_svc_->Set(&context, req, &resp);
+    absl::Time execute_time = absl::Now();
+
+    // RPC failed
+    if (!status.ok()) {
+      return MAKE_ERROR(ERR_INTERNAL)
+        << "gRPC Get call failed: "
+        << status.error_message();
+    }
+
+    int execute_duration =
+        (execute_time - start_time) / absl::Microseconds(1);
+
+    auto result_str = resp.DebugString();
+    if (result_str.size() <= 0) {
+      std::cout << "No Results" << std::endl;
+    } else {
+      std::cout << result_str << std::endl;
+    }
+    std::cout << "Executed query in " << execute_duration << " us."
+              << std::endl;
+    return ::util::OkStatus();
+  }
+
+  enum cmd_type {Get, Subscribe, Set};
+
+  // Runs the main CLI loop.
+  ::util::Status  RunCli() {
```

Comment:
remove one space between `Status` and `RunCli`

Reply:
done